### PR TITLE
Post to network upload model

### PR DESCRIPTION
**Pull request by jvfdunkley**
### :pushpin: References

-   **Issue:** Closes [Post to network upload model](https://app.clickup.com/t/gt7rkc)

### :memo: Implementation
- Sent over upload data to "http://rinchen-dos.cnb.csic.es:8882/upload" via a proxy and Axios using Future. It was sent as a multi-part form using the FormData object.
- Added a loading screen from Material UI because the request takes a while 

### :art: Screenshots
Screen Shot 2021-04-09 at 2 38 33 PM

### :bookmark_tabs: Others

-   [ ] Any change in the [API repo](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?
-   [ ] Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?

### CODE REVIEW

**Review by adrianq — APPROVED** [review]

**Comment by adrianq** [comment]
@jvfdunkley  All good with this PR. @tokland  I have merged this branch with yours. Feel free to review this PR and merge it whenever you are fine with it

File: app/assets/javascripts/3dbio_viewer/src/webapp/components/annotations-tool/AnnotationsTool.tsx [hunk 1]
```diff
@@ -54,17 +55,24 @@ export const AnnotationsTool: React.FC<AnnotationsToolProps> = React.memo(props
...
+            setError(
+                i18n.t("Error: Missing file - please upload an annotations file in JSON format.")
+            );
         } else {
             setError("");
             window.alert("TODO");
         }
     }, []);
 
+    const SwitchToggle = useCallback(() => {
```
Comment (tokland) [inline app/assets/javascripts/3dbio_viewer/src/webapp/components/annotations-tool/AnnotationsTool.tsx]: functions -> downcase.

File: app/assets/javascripts/3dbio_viewer/src/webapp/components/dropzone/Dropzone.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const Shade = styled.div<DropzoneRootProps>`
     display: flex;
     place-content: center;
     text-shadow: 1px 1px 10px black;
-    padding: 20px;
+    padding: 60px;
     align-items: center;
     border-width: 2px;
     border-radius: 2px;
```
Comment (tokland) [inline app/assets/javascripts/3dbio_viewer/src/webapp/components/dropzone/Dropzone.tsx]: I am moving this code to the bottom (this came from existing code), to keep props and main component on the top of the file and the first thing you see.

File: app/assets/javascripts/3dbio_viewer/src/webapp/components/model-upload/ModelUpload.tsx [hunk 3]
```diff
@@ -38,39 +41,42 @@ export const ModelUpload: React.FC<ModelUploadProps> = React.memo(props => {
...
+            setOpen(true);
+            return compositionRoot.uploadAtomicStructure(uploadParams).run(
+                result => {
+                    setAtomicStructure(result);
+                    setOpen(false);
+                    openUploadConfirmation();
+                },
+                error => {
+                    setOpen(false);
+                    return error;
```
Comment (tokland) [inline app/assets/javascripts/3dbio_viewer/src/webapp/components/model-upload/ModelUpload.tsx]: The return values of `Future.run` callbacks are ignored, so no need to return anything. It's the only point we work with side-effects. I've added `setError(error.message)`, does it make sense?

File: app/assets/javascripts/3dbio_viewer/src/webapp/components/upload-loader/UploadLoader.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+interface LoaderProps {
+    open: boolean;
+}
+
+export const UploadLoader: React.FC<LoaderProps> = React.memo(props => {
+    const classes = useStyles();
+    const { open } = props;
+    return (
+        <Backdrop className={classes.backdrop} open={open}>
```
Comment (tokland) [inline app/assets/javascripts/3dbio_viewer/src/webapp/components/upload-loader/UploadLoader.tsx:21]: Cool! We hadn't used yet this component.

**Review by tokland — APPROVED** [review]
:+1: Code and feature: All good! Just some minor comments/commits.

**Pull request merged**